README.md

```diff
@@ -327,6 +338,164 @@ Default: `{{ __mssql_client_repository }}`
 
 Type: `string`
 
+### `mssql_ha_configure`
+
+Use the variables with the `mssql_ha_` prefix to configure SQL Server for high
+availability.
+
+Set to `true` to configure for high availability. Setting to `false` does not
+remove configuration for high availability.
+
+When set to `true`, the role performs the following tasks:
+1. Include the fedora.linux_system_roles.firewall role to configure firewall:
+1.1. Open the firewall port set with the `mssql_ha_listener_port` variable.
+1.2. Enable the `high-availability` service in firewall.
+2. Configure SQL Server for high availability:
+2.1. Enable AlwaysOn Health events.
+2.2. Create certificate on the primary replica and distribute to other replicas.
+2.3. Configure endpoint and availability group.
+2.4. Configure the pacemaker user.
+3. Include the fedora.linux_system_roles.ha_cluster role to configure pacemaker.
+
+Default: `false`
+
+Type: `bool`
+
+#### `mssql_ha_replica_type`
+
+A host variable that specifies the type of the replica to be configured on this
+host.
+
+See [`Setting Up SQL Server and Configuring for High Availability`](#Setting-Up-SQL-Server-and-Configuring-for-High-Availability) for an example
+inventory.
+
+You must set the `mssql_ha_replica_type` variable to `primary` for exactly one
+host.
+
+The available values are: `primary`, `synchronous`, `witness`.
+
+Default: no
+
+Type: `str`
+
+#### `mssql_ha_firewall_configure`
+
+Whether to configure firewall for high availability or not.
+
+The role configures firewall only if the `firewalld` service has been started or
+enabled.
+
+Default: `true`
+
+Type: `bool`
+
+#### `mssql_ha_listener_port`
+
+The TCP port to be used for high availability.
+
+Default: `5022`
+
+Type: `int`
+
+#### `mssql_ha_cert_name`
+
+The name of the certificate used for high availability configuration.
+
+Default: `null`
+
+Type: `str`
+
+#### `mssql_ha_master_key_password`
+
+The password to set for the master key used with the certificate.
+
+Default: `null`
+
+Type: `str`
+
+#### `mssql_ha_private_key_password`
+
+The password to set for the private key used with the certificate.
+
+Default: `null`
+
+Type: `str`
+
+#### `mssql_ha_reset_cert`
+
+Whether to reset certificates used for high availability or not.
+
+Default: `false`
+
+Type: `bool`
+
+#### `mssql_ha_endpoint_name`
+
+The name of the endpoint to be configured.
+
+Default: `null`
+
+Type: `string`
+
+
+#### `mssql_ha_ag_name`
+
+The name of the availability group to be configured.
+
+Default: `null`
+
+Type: `string`
+
+#### `mssql_ha_db_name`
+
+The name of the database to be replicated. This database must exist in SQL
+Server.
+
+Default: `null`
+
+Type: `string`
+
+#### `mssql_ha_db_backup_path`
+
+The path to back up replicated database to.
```

Comment:
For SQL Server, any database participating in an Availability Group must be in full recovery mode and  have a valid log backup. This path will be used to backup the database prior to initiating replication within the Always On availability group.

Reply:
The role sets a full recovery mode and backs the database provided with `mssql_ha_db_backup_path` if no back-up newer than 3 hours exists. I need to mention this too in README for transparency.

Reply:
Applied